library/src/main/java/ai/promoted/metrics/id/AncestorId.kt

```diff
@@ -0,0 +1,48 @@
+package ai.promoted.metrics.id
+
+/**
+ * Represents any of the top-level "ancestor" IDs (i.e. session ID, view ID, logUserID) which
+ * can either be explicitly set by the library user, or implicitly set by calling [advance].
+ */
+internal class AncestorId(
+    idGenerator: IdGenerator
+) {
+    private val advanceableId = AdvanceableId(
+        skipFirstAdvancement = true,
+        idGenerator
+    )
+
+    private val pendingImplicitValue = advanceableId.currentValue
+
+    var isOverridden = false
+        private set
+
+    var currentValue = ""
+        private set
+
+    val pendingOrCurrentValue
```

Comment:
Nit: On iOS I named this `currentOrPendingValue`. Would be nice to line up the names. What do you think?